### PR TITLE
add domain options required for NUMA

### PR DESCRIPTION
**Pull request by antholeole**
some options that are required to generate a domain with numa

no tests but verified working in my own repo - i can add test if required

### CODE REVIEW

File: generate-xml/domain.nix [hunk 1]
```diff
@@ -39,6 +39,10 @@ let
             (subelem "node" [ (subattr "id" typeInt) (subattr "bandwidth" typeInt) ] [ ])
           ])
         ])
+        (subelem "numatune" [ ] [
+          (subelem "memory" [ (subattr "mode" typeString) (subattr "nodeset" typeString) ] [ ])
+          (subelem "memnode" [ (subattr "mode" typeString) (subattr "nodeset" typeString) (subattr "cellid" typeString) ]  [ ])
+        ])
```
Comment (AshleyYakeley) [inline generate-xml/domain.nix:45]: Should `numatune.memnode.cellid` be `typeInt` instead?

File: generate-xml/domain.nix [hunk 2]
```diff
@@ -195,6 +199,14 @@ let
               (subattr "policy" typeString)
               (subattr "name" typeString)
             ] [ ])
+            (subelem "numa" [ ] [
+                (subelem "cell" [
+                  (subattr "id" typeString)
```
Comment (AshleyYakeley) [inline generate-xml/domain.nix:204]: Should `numa.cell.id` be `typeInt` instead?

**Review by AshleyYakeley — CHANGES_REQUESTED** [review]

File: generate-xml/domain.nix (hunk 1 above)
Comment (antholeole) [reply generate-xml/domain.nix:45]: yes, I think so - I can't think of a case where cellid would not be an integer

**Comment by AshleyYakeley** [comment]
Thanks

**Pull request merged**